include/mbedtls/pk.h

```diff
@@ -397,7 +526,19 @@ int mbedtls_pk_encrypt( mbedtls_pk_context *ctx,
  * \param pub       Context holding a public key.
  * \param prv       Context holding a private (and public) key.
  *
- * \return          0 on success or MBEDTLS_ERR_PK_BAD_INPUT_DATA
+ * \return          * 0 on success.
+ *                  * MBEDTLS_ERR_PK_BAD_INPUT_DATA if one of the contexts
+ *                    is ill-formed.
+ *                  * MBEDTLS_ERR_PK_TYPE_MISMATCH if the contexts cannot
+ *                    represent keys of the same type.
+ *                  * MBEDTLS_ERR_PK_FEATURE_UNAVAILABLE if it is impossible
+ *                    to determine whether the keys match. This can only happen
+ *                    if \c prv is an opaque key.
+ *                  * Or a type-specific error code.
+ *
+ * \note            Opaque key types may omit implementing this function
+ *                  by providing a NULL pointer in the mbedtls_pk_info_t structure.
+ *                  An opaque \c pub never matches a transparent \c prv.
```

Comment:
@gilles-peskine-arm Why would we enforce that here?

Reply:
It's not so much that we actively enforce it, rather we don't do anything to support matching an opaque public key with a transparent private key, and so they won't match. Maybe we should document that as a current limitation rather than as a guarantee, but I doubt that this limitation will ever be lifted, I don't see any use case for it.

Reply:
I think as long as we are not fully sure that this will never be sensible, we should not restrict ourselves unnecessarily by publicly documenting this limitation. My preference would be to remove the comment altogether, giving us more flexibility for future changes.

Reply:
I do want to document that this is the current behavior. We could add “currently” to the sentence.

Reply:
@gilles-peskine-arm Ok, let's do that. @AndrzejKurek Are you going to change this?

Reply:
Yes, I'm just waiting for a conclusion on other questions, so that I can collect all the relevant changes into one PR.